src/Microsoft.Health.Fhir.Api/Controllers/ExportController.cs

```diff
@@ -0,0 +1,131 @@
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.Health.Fhir.Api.Features.ActionResults;
+using Microsoft.Health.Fhir.Api.Features.Filters;
+using Microsoft.Health.Fhir.Api.Features.Routing;
+using Microsoft.Health.Fhir.Core.Configs;
+using Microsoft.Health.Fhir.Core.Exceptions;
+using Microsoft.Health.Fhir.Core.Features.Context;
+
+namespace Microsoft.Health.Fhir.Api.Controllers
+{
+    [ServiceFilter(typeof(OperationOutcomeExceptionFilterAttribute))]
+    public class ExportController : Controller
+    {
+        /*
+         * We are currently hardcoding the routing attribute to be specific to Export and
+         * get forwarded to this controller. As we add more operations we would like to resolve
+         * the routes in a more dynamic manner. One way would be to use a regex route constraint
+         * - eg: "{operation:regex(^\\$([[a-zA-Z]]+))}" - and use the appropriate operation handler.
+         * Another way would be to use the capability statement to dynamically determine what operations
+         * are supported.
+         * It would be easier to determine what pattern to follow once we have built support for a couple
+         * of operations. Then we can refactor this controller accordingly.
+         */
+
+        private readonly ILogger<ExportController> _logger;
+        private readonly IFhirRequestContextAccessor _fhirRequestContextAccessor;
+        private readonly ExportConfiguration _exportConfig;
+
+        public ExportController(
+            ILogger<ExportController> logger,
+            IFhirRequestContextAccessor fhirRequestContextAccessor,
+            IOptions<ExportConfiguration> exportConfig)
+        {
+            EnsureArg.IsNotNull(logger, nameof(logger));
+            EnsureArg.IsNotNull(fhirRequestContextAccessor, nameof(fhirRequestContextAccessor));
+            EnsureArg.IsNotNull(exportConfig, nameof(exportConfig));
+
+            _logger = logger;
+            _fhirRequestContextAccessor = fhirRequestContextAccessor;
+            _exportConfig = exportConfig.Value;
+        }
+
+        [HttpGet]
+        [Route(KnownRoutes.Export)]
+        [ValidateExportHeadersFilter]
+        public IActionResult Export()
+        {
+            HttpStatusCode returnCode;
+            OperationOutcome result;
+
+            if (_exportConfig.Enabled)
+            {
+                result = GenerateOperationOutcome(
+                    OperationOutcome.IssueSeverity.Error,
+                    OperationOutcome.IssueType.NotSupported,
+                    Resources.NotFoundException);
+
+                returnCode = HttpStatusCode.NotImplemented;
+            }
+            else
+            {
+                throw new RequestNotValidException(string.Format(Resources.UnsupportedOperation, "Export"));
+            }
+
+            return FhirResult.Create(result, returnCode);
+        }
+
+        [HttpGet]
+        [Route(KnownRoutes.ExportResourceType)]
```

Comment:
Shouldn't these method also check to see if the config is enabled or not?

Reply:
These two just validate whether the `type` is expected `ResourceType` and then just call the base `Export()` method, which has the logic to respond based on whether export is enabled or not. I have moved the logic to a separate method and am calling that method from each action.